src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -166,23 +170,59 @@ private void produce(Object[] r) throws IOException, KettleStepException
         }
     }
 
-    private void consume() throws IOException, KettleStepException
+    private void consume() throws IOException, KettleStepException, InterruptedException
     {
-        GetResponse response;
-
+        GetResponse response 		= null;
+	QueueingConsumer consumer 	= null;
+        if (data.isWaitingConsumer) { 
+		logMinimal("WAITING FOR MESSAGES");
+		consumer = new QueueingConsumer(channel) {
+		   @Override
+		  public void handleCancel(String consumerTag) throws IOException {
+			logBasic(consumerTag + "Canceled");
+		  }
+		  public void handleShutdownSignal(String consumerTag, ShutdownSignalException sig) {
+			logDebug(consumerTag + " :SHUTDOWN: " + sig.getMessage() );
+		  }
+		};
+		channel.basicConsume(data.target, false, consumer);
+	}
         do {
-            response = channel.basicGet(data.target, false);
+	    if (data.isWaitingConsumer ) {
+		    QueueingConsumer.Delivery delivery = null;
+		    while (!isStopped() && delivery == null && !isStepTimedoutAsConsumer() ) {  delivery = consumer.nextDelivery(10000); } ;

```

Comment:
| I was wondering if we should create a new tab for bindings, know do you think ?
yes, we should just don't have time to move them to separate tab.